### PR TITLE
Update Remote Config real-time config update ref-docs

### PR DESCRIPTION
**Pull request by qdpham13**
Update Remote Config real-time reference docs to be more description and uniform. #no-changelog

### CODE REVIEW

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (danasilver) [inline FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:44]: nit: can remove "being"

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (morganchen12) [inline FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:176]: nit: `anytime` -> `whenever`, `their` -> `there`

**Review by morganchen12 — COMMENTED** [review]
LGTM, but please wait for Dana's approval.

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (qdpham13) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h]: Updated

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (qdpham13) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h]: Updated

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (danasilver) [inline FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:331]:
```suggestion
#pragma mark - Real time Config Updates
```

Comment (danasilver) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:331]: Realized later today there's no hyphen in this use case.

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (danasilver) [inline FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:176]:
```suggestion
/// `FIRRemoteConfigUpdateCompletion` when a new config version has been automatically fetched
```

Comment (danasilver) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:176]: I think auto-fetch might be too internal to the SDKs to use in the docs

Comment (danasilver) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:176]: This is a great descriptive comment, btw!

**Review by danasilver — APPROVED** [review]
Left a couple suggestions

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (danasilver) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:331]: Retracting :P

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (danasilver) [inline FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:355]: :) nice catch

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (danasilver) [inline FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:336]: Is it possible to link to `FIRRemoteConfigUpdate` here so folks have an easy link to the information? cc @morganchen12 is there `@link` syntax here?

Comment (morganchen12) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:336]: You can put a link directly into the header.

Comment (qdpham13) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h]: Updated the comment. I think it should now have a link to the pragma mark for FIRRemoteConfigUpdate

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (qdpham13) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h]: Updated

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (morganchen12) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h]: I don't see the link here?

Comment (qdpham13) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h]: Instead of the link I'm using the double back-ticks to reference the `FIRRemoteConfigUpdate` method: https://stackoverflow.com/questions/38321880/how-to-go-about-adding-a-link-reference-to-another-method-in-documentation-xcode

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (danasilver) [inline FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:335]: If double backticks don't link to the method in our docs, I'd just use single so it renders as code.

```
@param configUpdate An instance of `FIRRemoteConfigUpdate` with information on which keys' values have changed.
```

**Comment by danasilver** [comment]
LGTM pending the backtick linking discussion

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (qdpham13) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h]: @morganchen12 can you confirm if the double back-ticks link to the method when the docs are generated?

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h
Comment (qdpham13) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h]: Changing to single back-ticks for now

**Pull request merged**
